File: tiq-e/array/4/main.py
from time import perf_counter
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        s = set(nums)
        return len(s) != len(nums)


arr = [1, 2, 3, 4, 5]

start = perf_counter()
result = Solution().containsDuplicate(arr)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/array/4/main.rs
use std::collections::HashSet;
use std::iter::FromIterator;
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let set: HashSet<i32> = HashSet::from_iter(nums.clone());
        if set.len() != nums.len() {
            return true;
        }
        false
    }
}

fn main() {
    let arr = Vec::from([1, 2, 3, 3]);

    let start = Instant::now();
    let result = Solution::contains_duplicate(arr);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}